surefire-provider/src/main/java/org/arquillian/smart/testing/surefire/provider/TestStrategyApplier.java

```diff
@@ -44,18 +48,29 @@ TestsToRun apply(Configuration configuration) {
                 + "For details on how to configure it head over to http://bit.ly/st-config");
             return Collections.emptySet();
         }
-        final Set<TestSelection> selectedTests = new LinkedHashSet<>();
+        final List<TestSelection> selectedTests = new ArrayList<>();
```

Comment:
Is it really necessary to have it as a `List`? With `Set` we were implicitly removing duplicates. Does it still hold true?